api/python/quilt3/packages.py

```diff
@@ -1448,6 +1494,15 @@ def selector_fn(*args):
                     f"'build' instead."
                 )
 
+        if selector_fn is None:
+            # When pushing to S3, do not copy files if they are in the same
+            # bucket as the destination registry.
+            def selector_fn(logical_key, entry):
+                if not registry_parsed.is_local():
+                    if entry.physical_key.bucket == registry_parsed.bucket:
+                        return False
+                return True
```

Comment:
AFAICT we only allow push to S3 bucket so it might make sense to do this

```suggestion
        assert not registry_parsed.is_local()

        if selector_fn is None:
            # Do not copy files if they are in the same bucket as the destination registry.
            def selector_fn(logical_key, entry):
                return entry.physical_key.bucket != registry_parsed.bucket
```